httpserver/service_node.cpp

```diff
@@ -85,20 +133,28 @@ ServiceNode::ServiceNode(boost::asio::io_context& ioc, uint16_t port,
 
 ServiceNode::~ServiceNode() = default;
 
-void ServiceNode::relay_one(const message_t& msg, sn_record_t sn) const {
-
-    BOOST_LOG_TRIVIAL(debug) << "Relaying a message to " << sn;
-
-    request_t req;
-    serialize_message(req.body(), msg);
+void ServiceNode::relay_one(const std::shared_ptr<request_t>& req,
+                            sn_record_t sn) const {
 
-    req.target("/v1/swarms/push");
+    BOOST_LOG_TRIVIAL(debug) << "Relaying a message to: " << sn;
 
+    // TODO: consider storing a shared_ptr inside http session instead of a copy
+    // (that might be annoying for requests that don't want to create a
+    // shared_ptr, like swarm updates)
     make_http_request(
-        ioc_, sn.address, sn.port, req, [this, sn](sn_response_t&& res) {
+        ioc_, sn.address, sn.port, *req, [this, sn, req](sn_response_t&& res) {
```

Comment:
Maybe we could explore the option to have the signature of `http_callback_t` provide both the request and the response objects?
I might have seen that somewhere in beast.
Then only `make_http_request` would be responsible for taking the request, and once it's finished it would pass it on (move) to the callback.
Worth exploring you think?

Reply:
Not sure if it is. What problem will it solve? If it is about making fewer copies, I don't think it will be a concert in the future. I'm currently working on similar PR for batches and it seems that passing a shared_ptr into http_requests is not an option there (instead of making a copy).

Reply:
Fair enough, I was just trying to see if that would simplify things (in terms of object lifetime and ownership).